packages/core/src/index.js

```diff
@@ -735,6 +735,27 @@ class Jimp extends EventEmitter {
   }
 
   setPixelColour = this.setPixelColor;
+
+  /**
+   * Determine if the image contains opaque pixels.
+   * @return {boolean} hasAlpha whether the image contains opaque pixels
+   */
+  hasAlpha() {
+    let hasAlpha = false;
+
+    for (let yIndex = 0; yIndex < this.bitmap.height && !hasAlpha; yIndex++) {
+      for (let xIndex = 0; xIndex < this.bitmap.width && !hasAlpha; xIndex++) {
+        const idx = (this.bitmap.width * yIndex + xIndex) << 2;
+        const alpha = this.bitmap.data[idx + 3];
+
+        if (alpha !== 0xff) {
+          hasAlpha = true;
```

Comment:
I'm not sure how well v8 and others read the code ahead, but wouldn't it be faster to return early ?

Reply:
the for loops check for hasAlpha to be false so once if find an alpha channel both the loops break. I will go through every pixel if it doesn't find an alpha channel though.